src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -480,16 +490,34 @@ protected void parseCreateField(ParseContext context, int dimension) throws IOEx
         validateIfKNNPluginEnabled();
         validateIfCircuitBreakerIsNotTriggered();
 
-        Optional<float[]> arrayOptional = getFloatsFromContext(context, dimension);
+        if (VectorDataType.BYTE.equals(vectorDataType)) {
+            Optional<byte[]> bytesArrayOptional = getBytesFromContext(context, dimension);
 
-        if (!arrayOptional.isPresent()) {
-            return;
+            if (!bytesArrayOptional.isPresent()) {
+                return;
+            }
+            final byte[] array = bytesArrayOptional.get();
+            VectorField point = new VectorField(name(), array, fieldType);
+
+            context.doc().add(point);
+            addStoredFieldForVectorField(context, fieldType, name(), point.toString());
+        } else if (VectorDataType.FLOAT.equals(vectorDataType)) {
+            Optional<float[]> floatsArrayOptional = getFloatsFromContext(context, dimension);
+
+            if (!floatsArrayOptional.isPresent()) {
+                return;
+            }
+            final float[] array = floatsArrayOptional.get();
+            VectorField point = new VectorField(name(), array, fieldType);
+
+            context.doc().add(point);
+            addStoredFieldForVectorField(context, fieldType, name(), point.toString());
```

Comment:
This section also looks like a good candidate for being implemented in the type specific enum member.

Reply:
I already tried to do this but the `getBytesFromContext()` is in KNNVectorFieldMapper class and we can't call this method from enum class